src/mongocrypt-ciphertext.c

```diff
@@ -159,7 +163,7 @@ _mongocrypt_ciphertext_serialize_associated_data (
 {
    int32_t bytes_written = 0;
 
-   if (!out) {
+   if (!ciphertext || !out) {
```

Comment:
Suggest asserting `ciphertext` as not-null instead.

Reply:
:+1: Change coming in next commit.